test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/JsonNoMetadataLevelTests.cs

```diff
@@ -13,14 +13,21 @@ namespace Microsoft.OData.Tests.JsonLight
 {
     public class JsonNoMetadataLevelTests
     {
-        private readonly JsonNoMetadataLevel testSubject = new JsonNoMetadataLevel();
+        private readonly JsonNoMetadataLevel testSubject = new JsonNoMetadataLevel(/* alwaysAddTypeAnnotationsForDerivedTypes */ false);
 
         [Fact]
         public void NoMetadataLevelShouldReturnNoMetadataTypeOracleWhenKnobIsSet()
         {
             Assert.IsType<JsonNoMetadataTypeNameOracle>(testSubject.GetTypeNameOracle());
         }
 
+        [Fact]
+        public void NoMetadataLevelShouldReturnMinimalMetadataTypeOracleWhenKnobIsSet()
+        {
+            JsonNoMetadataLevel testSubjectWithTypeAnnotations = new JsonNoMetadataLevel(/* alwaysAddTypeAnnotationsForDerivedTypes */ true);
+            Assert.IsType<JsonMinimalMetadataTypeNameOracle>(testSubject.GetTypeNameOracle());
+        }

```

Comment:
can merge these two cases using 
[Theory]
[InlineData(false, typeof(JsonNoMetadataTypeNameOracle))
[InlineData(ture, typeof(JsonMinimalMetadataTypeNameOracle))

Reply:
Tests merged in the 4th commit to this PR.